crates/runtime/src/dataconnector/odbc.rs

```diff
@@ -85,20 +109,33 @@ where
         }
 
         Box::pin(async move {
-            let driver = params
-                .get("odbc_connection_string")
-                .context(MissingParameterSnafu {
-                    param: "odbc_connection_string".to_string(),
-                })?
-                .expose_secret()
-                .to_lowercase();
-
-            let driver = driver
-                .split(';')
-                .find(|s| s.starts_with("driver="))
-                .context(NoDriverSpecifiedSnafu)?;
-
-            let dialect = match_driver_to_dialect(driver);
+            let dialect = if let Some(sql_dialect) = params.get("sql_dialect") {
+                let driver: ODBCDriver = ODBCDriver::from(sql_dialect.expose_secret().as_str());
```

Comment:
This is overloading the sql dialect with the ODBC driver - it would be cleaner to have them separate and be able to convert from the sql dialect to the ODBC driver. i.e. I could imagine we want to support a "generic" dialect, but that wouldn't ever be a ODBC driver